### PR TITLE
Clean up rewards and delegation management when empty

### LINKED ISSUE #1136: Delegations & rewards not cleaned up after undelegated
**Describe the bug**
* Removing a delegation does not eventually remove the delegator from the list, even after unbonded.
* Removing a delegation claims rewards but the rewards aren't removed from the staking rewards tab.

**Environment and Version (please complete the following information):**
 - OS: [e.g. Windows, Mac, Linux] Mac
 - Version [e.g. v0.0.1] v0.8.1

**To Reproduce**
1. Undelegate your CRO. Wait the 28 days.
2. Notice the validator still shows up in the list with an 'undelegate' and 'redelegate' stake option.
3. Check the staking rewards tab. It still shows you can withdraw awards from the delegator. Attempting to withdraw these will fail (because they don't exist)

**Expected behavior**
* Expect delegators to be cleaned up from the delegation management list if no longer in-use
* Expect rewards to be cleared if they were already claimed

**Screenshots**
Screen Shot 2022-06-21 at 10 22 12 PM
Screen Shot 2022-06-21 at 10 24 01 PM

### PR DESCRIPTION
**Pull request by jlbernal**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

Fixes #1136 

There was already a check for an empty delegation or reward response. Previously they would just return an empty promise. I reused the storage deletion code to now clean up rewards or delegations before returning the promise if the responses are empty.

# PR Checklist:

- [x] Have you read the [CONTRIBUTING.md](/CONTRIBUTING.md)?
- [x] Have you rebased your work on top of the latest `dev`? 
- [x] Does your PR follow the [Conventional Commits Guide](https://www.conventionalcommits.org/)?
- [x] Have you checked your code compiles? (`yarn build`)
- [ ] Have you included tests for any non-trivial functionality?
- [x] Have you followed the [Multi-Language Content Management Guide](/CONTENT_MANAGEMENT.md)?
- [x] Have you checked your code passes the unit tests? (`yarn test`)
- [x] Have you checked your code formatting is correct? (`yarn lint:js`)
- [x] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`yarn audit`)
- [x] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](/CONTRIBUTING.md#developer-certificate-of-origin).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Comment by jlbernal** [comment]
This is my first contribution to this project and state my PR is in accordance with the Developer's Certificate of Origin

File: src/service/storage/StorageService.ts [hunk 1]
```diff
@@ -386,15 +405,7 @@ export class StorageService {
...
-    await this.db.commonTransactionStore.remove(
-      {
-        walletId: rewardTransactions.walletId,
-        txType: 'reward',
-      },
-      {
-        multi: true,
-      },
-    );
+    await this.removeRewardList();
```
Comment (crypto-matto) [inline src/service/storage/StorageService.ts:408]:
```suggestion
    await this.removeRewardList(rewardTransactions.walletId);
```

File: src/service/storage/StorageService.ts [hunk 2]
```diff
@@ -361,9 +367,22 @@ export class StorageService {
     return this.db.stakingStore.insert(stakingTransactions); */
   }
 
+  public async removeRewardList() {
+    return this.db.commonTransactionStore.remove(
+      {
```
Comment (crypto-matto) [inline src/service/storage/StorageService.ts:372]:
```suggestion
      {
          walletId
```

File: src/service/storage/StorageService.ts [hunk 3]
```diff
@@ -361,9 +367,22 @@ export class StorageService {
     return this.db.stakingStore.insert(stakingTransactions); */
   }
 
+  public async removeRewardList() {
```
Comment (crypto-matto) [inline src/service/storage/StorageService.ts:370]:
```suggestion
  public async removeRewardList(walletId: string) {
```

**Review by crypto-matto — COMMENTED** [review]
Hi @jlbernal, my pleasure to have you submitted a first PR to Desktop Wallet & fixing an issue!  Much appreciated! 

Sorry for replying you so late as we're moving on to publish a next release soon so that's why we are reviewing the public PR until recently. 

Your reported issue is identified & making much sense. Reviewing through the Staking module and we do dig out more bugs. We reviewed & tested out your codes, seeing that the PR can be done more. We're happy to help, guide you through, and test together. See if you want to offer more help & polish your first PR better! 

## Similar Issues
1. Every un-delegate transaction will also withdraw the corresponding rewards together. Yet after an un-delegation, the previous reward is still there. That would be even more confusing when there's only one last delegation left to be un-delegated. It's suggested that we may also apply a `fetchAndSaveRewards()` in `sendUnDelegateTransaction()` so that the reward list can be re-fetched again. 
2. We also find that occasionally, the list of rewards or delegation are duplicated.

**Review by crypto-matto — APPROVED** [review]
Thanks for your PR @jlbernal. We're merging some of your work into the next release. 👍🏻

**Comment by jlbernal** [comment]
Thank you for improving and merging my PR! Sorry I didn't get back earlier, been busy with work. I would have looked at the other bugs you mentioned too.

I like this wallet so I'll keep an eye out for other small contributions I could make in the future. Thanks again!

**Pull request merged**